test/performance/README.md

```diff
@@ -0,0 +1,125 @@
+a command line.
+
+```shell
+make bin
+./bin/antrea-scale --kubeConfigPath=/root/.kube/config --timeout=120 --config=./test/performance/scale.yml
+```
+
+## 2. Configurability
+
+You can easily configure the test parameters to meet various testing scenarios through a YAML file.
+
+```yaml
+real_node: true
+repeat_times: 1
+namespace_num: 5
+pods_num_per_ns: 10
+svc_num_per_ns: 4
+np_num_per_ns: 5
+skip_deploy_workload: false
+teardown: false
+scales:
+  - name: "ScaleUpWorkloadPods"
+    package: "test/performance/framework"
+    repeat_times: 1
+  - name: "ScaleService"
+    package: "test/performance/framework"
+    repeat_times: 1
+  - name: "ScaleRestartAgent"
+    package: "test/performance/framework"
+    repeat_times: 1
+  - name: "RestartController"
+    package: "test/performance/framework"
+    repeat_times: 1
+  - name: "ScaleNetworkPolicy"
+    package: "test/performance/framework"
+    repeat_times: 1
+```
+
+## 3. Flexibility & Assemblability
+
+The configuration file can be divided into two parts:
+the first part controls the scale of the test,
+
+```yaml
+real_node: true
+repeat_times: 1
+namespace_num: 5
+pods_num_per_ns: 10
+svc_num_per_ns: 4
+np_num_per_ns: 5
+skip_deploy_workload: false
+teardown: false
+```
+
+and the second part controls different test cases through the combination and arrangement of
+individual test cases.
+
+```yaml
+scales:
+  - name: "ScaleNetworkPolicy"
+    package: "test/performance/framework"
+    repeat_times: 1
+  - name: "ScaleRestartAgent"
+    package: "test/performance/framework"
+    repeat_times: 1
+```
+
+For example, if we want to test whether the startup speed of the agent is affected after creating
+NetworkPolicy on a large scale, we can place "ScaleNP" under "ScaleAgent."
+
+Also, we can control the number of tests by setting the "repeat" parameter.
+
+## 4. Efficient resource utilization(Antrea simulator agent)
+
+A significant concern is that large-scale testing requires a vast amount of cluster resources.
+We can use simulated agents to conduct tests to save resources and achieve the goal of scaling
+tests with fewer resources.
+
+If we want to test the performance of the Antrea Controller with limited node resources, we can
+use the "realNode" parameter. Once disabled, the system will use simulated Antrea agent.
+The simulator can watch the Antrea controller just like the real agent does, and it makes us
+able to simulate a large number of agents in a smaller number of nodes. It is useful for Antrea
+scalability testing, without having to create a very large cluster.
+
+## 5. Multiple platforms(Different CNIs)
+
+Additionally, for some common functional features, it's essential to compare the performance
+differences between different CNIs to understand our shortcomings or advantages.
+The scale test tool can run tests on different Kubernetes platforms and compare the performance
+metrics of different CNIs.
```

Comment:
Have you tried with different CNIs? If yes, please highlight the configuration yaml differences and requirements, I don't think the case like `ScaleRestartAgent` will work for other CNIs. If not, please remove this section.

Reply:
Of course. I think we can use `ScaleService` and `ScaleNetworkPolicy` to test any other CNIs.
```yaml
scales:
  - name: "ScaleUpWorkloadPods"
    package: "test/performance/framework"
    repeat_times: 1
  - name: "ScaleService"
    package: "test/performance/framework"
    repeat_times: 1
  - name: "ScaleNetworkPolicy"
    package: "test/performance/framework"
    repeat_times: 1
```